### PR TITLE
Source File: Update Dockerfile to latest Airbyte template

### PR DESCRIPTION
**Pull request by Phlair**
## What
As a POC part of https://github.com/airbytehq/airbyte/issues/6455, I'm going to have source-file-secure use the new intra-repo dependency method to inherit from source-file.

This PR simply updates source-file's Dockerfile to our current layout so that this will actually work. This is because the intra-repo dependency method (at least the template example) relies on py libs & module code files from the dependency module being in a particular directory in its docker image.

note: this isn't going to be a necessity, the paths can be changed to suit the dependency docker image locations, however I want this as a look-at-me example in the docs so best to use standard layout.

also, updated the spec.json so it passes acceptance tests

### CODE REVIEW

**Review by sherifnada — APPROVED** [review]
I would leave comments in the Dockerfile explaining why we have this unique setup there and not in other connectors (may end up being a link to the docs you eventually write) 

the alternative here is to leave the dockerfile as is, do what file secure is currently doing, but that limits us to only one such intra repo dependency right?  I think the explicit approach is better fwiw

**Comment by Phlair** [comment]
@sherifnada good shouts

> I would leave comments in the Dockerfile explaining why we have this unique setup 
I'll do that in the files-secure Dockerfile since that will be the one with the extra dependency lines. This one is now just matching our general format.

> but that limits us to only one such intra repo dependency right?
exactly, which is fine and functional in this case as is, but I'm altering these to create an example for the new method (allowing multi) which I can link to in the docs.

File: airbyte-integrations/connectors/source-file/Dockerfile [hunk 1]
```diff
@@ -1,16 +1,21 @@
-FROM python:3.9-slim
+FROM python:3.9-slim as base
+FROM base as builder
 
-# Bash is installed for more convenient debugging.
-RUN apt-get update && apt-get install -y jq curl bash && rm -rf /var/lib/apt/lists/*
+RUN apt-get update
+WORKDIR /airbyte/integration_code
+COPY setup.py ./
+RUN pip install --prefix=/install .
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-file/Dockerfile:7]: Can't we just install this in /usr/local instead of doing the whole multi stage thing?

Comment (Phlair) [reply airbyte-integrations/connectors/source-file/Dockerfile:7]: I've emulated [our current generator template](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/source-python/Dockerfile).

afaia, the [benefit to multi-staging](https://docs.docker.com/develop/develop-images/multistage-build/) like this is that any unneeded build dependencies, artifacts aren't in the final Docker image, potentially reducing image size (although I haven't tested how much difference it's actually making in the Airbyte case)

Comment (sherifnada) [reply airbyte-integrations/connectors/source-file/Dockerfile:7]: Oh ok

**Pull request merged**